src/test/run-make/incr-prev-body-beyond-eof/Makefile

```diff
@@ -0,0 +1,16 @@
+-include ../../run-make-fulldeps/tools.mk
+
+# Tests that we don't ICE during incremental compilation after modifying a
+# function span such that its previous end line exceeds the number of lines
+# in the new file, but its start line/column and length remain the same.
+
+SRC=$(TMPDIR)/src
+INCR=$(TMPDIR)/incr
+
+all:
+	mkdir $(SRC)
+	mkdir $(INCR)
+	cp a.rs $(SRC)/main.rs
+	$(RUSTC) -C incremental=$(INCR) $(SRC)/main.rs
+	cp b.rs $(SRC)/main.rs
+	$(RUSTC) -C incremental=$(INCR) $(SRC)/main.rs
```

Comment:
Hmm, we do have a specialized incremental compilation testing harness, but it relies on `#[cfg]` testing. I guess it may not be suitable for running this test. That said, I can't quite tell what would cause this test to fail -- did it used to ICE?

Reply:
Yes, just double-checked that this ICEs without the changes in this PR.

I was kind of leaning on reviewers to be able to tell me if there's a better way of doing this. :) The important thing is that the span of interest is the same in all ways except end line/col in both compilations (i.e. file name, start line/col, and length). I'm not sure that's possible to do with the other testing methods, but that's mostly on account of being new. :)

Reply:
Given that the start line/column must match i'm not sure if there would be a way to use the other testing harness.